crates/wysiwyg/src/dom/nodes/mention_node.rs

```diff
@@ -222,42 +281,25 @@ where
         buffer: &mut S,
         _: &MarkdownOptions,
     ) -> Result<(), MarkdownError<S>> {
-        use MentionNodeKind::*;
-
-        // There are two different functions to allow for fact one will use mxId later on
-        match self.kind() {
-            MatrixUrl { .. } => {
-                fmt_user_or_room_mention(self, buffer)?;
-            }
-            AtRoom => {
-                fmt_at_room_mention(self, buffer)?;
-            }
-        }
-
+        fmt_mention(self, buffer)?;
         return Ok(());
 
         #[inline(always)]
-        fn fmt_user_or_room_mention<S>(
+        fn fmt_mention<S>(
             this: &MentionNode<S>,
             buffer: &mut S,
         ) -> Result<(), MarkdownError<S>>
         where
             S: UnicodeString,
         {
-            // TODO make this use mxId, for now we use display_text
-            buffer.push(this.display_text());
-            Ok(())
-        }
+            let text = match this.kind() {
+                // for User/Room type, we use the mx_id in the md output
+                MentionNodeKind::User { mention }
+                | MentionNodeKind::Room { mention } => S::from(mention.mx_id()),
```

Comment:
For a user mention, the Matrix spec says we should use the potentially ambigious display name (if available) and only fall back to the mx ID when this is not available.

If we are concerned about being able to disambiguate the user given a Matrix message, this should still be possible by looking at the formatted HTML which contains the full mention URL.

Reply:
I think the long term plan to disambiguate at app level is https://github.com/matrix-org/matrix-spec-proposals/pull/3952 

Otherwise at the RTE level we will always parse the HTML so it will be fine, and as previously mentioned we need a custom Markdown to handle plain text mode with pills.

Reply:
Done in [a2e3361](https://github.com/matrix-org/matrix-rich-text-editor/pull/719/commits/a2e336146bac9ba49d380c279f7813345508f66d)